dashboard/app/helpers/student_work_helper.rb

```diff
@@ -0,0 +1,37 @@
+# This module contains helper methods for retrieving and evaluating student work.
+module StudentWorkHelper
+  def self.get_student_code(user_id, level, unit_id, code_version = nil)
+    s3 = Aws::S3::Client.new
+    bucket = CDO.sources_s3_bucket
+    base_dir = CDO.sources_s3_directory
+
+    storage_id = storage_id_for_user_id(user_id)
+    storage_app_id = get_storage_app_id(storage_id, user_id, level, unit_id)
+    user_level = UserLevel.where(user_id: user_id, level_id: level.id, script_id: unit_id).last
```

Comment:
- [ ] remove. if we know we have a user_level because that's the top of the funnel, no need to check again